### PR TITLE
Add new Discussions plugin

### PR DESCRIPTION
**Pull request by lowlighter**
Add new discussion plugin to display how many discussion user has started, and how many comments and answers provided.
Also track most posted categories
lowlighter

### CODE REVIEW

File: source/plugins/discussions/index.mjs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          do {
+            console.debug(`metrics/compute/${login}/discussions > retrieving discussions after ${cursor}`)
+            const {user:{repositoryDiscussions:{edges = [], nodes = []} = {}}} = await graphql(queries.discussions.categories({login, after:cursor ? `after: "${cursor}"` : ""}))
+            cursor = edges?.[edges?.length - 1]?.cursor
+            fetched.push(...nodes)
+            pushed = nodes.length
+            console.debug(`metrics/compute/${login}/discussions > retrieved ${pushed} discussions after ${cursor}`)
+          } while ((pushed) && (cursor))
+
+          //Compute favorite categorie
```
Comment (Nixinova) [inline source/plugins/discussions/index.mjs:32]:
```suggestion
          //Compute favorite category
```

Comment (lowlighter) [reply source/plugins/discussions/index.mjs]: Thanks! 
Seems I still can't remember the correct spelling each time 😅

File: source/plugins/discussions/index.mjs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        //Load started discussions
+        {
+          const fetched = []
+          const categories = {}
+          let cursor = null
+          let pushed = 0
+          do {
+            console.debug(`metrics/compute/${login}/discussions > retrieving discussions after ${cursor}`)
+            const {user:{repositoryDiscussions:{edges = [], nodes = []} = {}}} = await graphql(queries.discussions.categories({login, after:cursor ? `after: "${cursor}"` : ""}))
+            cursor = edges?.[edges?.length - 1]?.cursor
```
Comment (spenserblack) [inline source/plugins/discussions/index.mjs:26]: Super minor nitpick, but I think
```js
edges?.[edges?.length - 1]
```
has a slightly redundant check. If `edges == null`, the expression inside the brackets shouldn't be evaluated.
For example
```js
const foo = void 0;
foo?.[foo.bar] // undefined, doesn't throw error
```

Comment (lowlighter) [reply source/plugins/discussions/index.mjs:26]: Yes I think you're right 🙂 

Eventually I'd like to move to [`Array.at`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at) with `edges.at(-1)?.cursor` which seems cleaner, but I'm not sure that both node used version and vercel's one currently support it

Comment (Nixinova) [reply source/plugins/discussions/index.mjs:26]: Array#at was added to Node literally yesterday.

**Comment by Nixinova** [comment]
Can discussions be moved to be under followup? Thats a more natural position than at the end.

**Comment by lowlighter** [comment]
Yes it would makes more sense. 
You can patch [source/templates/classic/partials/_.json](https://github.com/lowlighter/metrics/blob/master/source/templates/classic/partials/_.json) if you want 👍

**Pull request merged**